qiime2/core/archive/archiver.py

```diff
@@ -227,6 +228,30 @@ def _as_zip_path(self, path):
         return path
 
 
+class _NoOpArchive(_Archive):
+    """For dealing with unzipped artifacts"""
+
+    @classmethod
+    def is_archive_type(cls, path):
+        return os.path.isdir(str(path))
+
+    def relative_iterdir(self, relpath=''):
+        seen = set()
+        for name in os.listdir(str(self.path)):
+            if name.startswith(relpath) and name not in seen:
+                seen.add(name)
+                yield name
+
+    def open(self, relpath):
+        return open(os.path.join(self.path, self.uuid, relpath))
+
+    def mount(self, filepath):
+        shutil.copytree(self.path, filepath, dirs_exist_ok=True)
+        root = pathlib.Path(os.path.join(filepath, str(self.uuid)))
```

Comment:
Hmm, actually we may not want to use `shutil.copytree` here as it would defeat the goal of the upcoming cache. I'll leave this PR open for now. But otherwise looks good!

Reply:
So we probably want to be able to mount things in place?

Reply:
yep!

Reply:
Sounds good. I made an `artifact_cache` branch off the latest master and merged this branch into it. If I make changes to this code it will probably be there now if that works.

Reply:
https://github.com/qiime2/qiime2/pull/623